src/main/scala/io/phdata/retirementage/YamlProtocols.scala

```diff
@@ -30,8 +30,52 @@ object YamlProtocols extends DefaultYamlProtocol {
   implicit val relatedTable: YamlFormat[ChildTable] = lazyFormat(yamlFormat5(ChildTable))
   implicit val hold                                 = yamlFormat3(Hold)
 
+  implicit val filterFormat: YamlFormat[CustomFilter] = lazyFormat(yamlFormat1(CustomFilter))
+  implicit val customTable: YamlFormat[CustomTable]   = lazyFormat(yamlFormat5(CustomTable))
+  implicit val tableFormat: YamlFormat[Table]         = TableYamlFormat
+
   implicit val databaseFormat = yamlFormat2(Database)
   implicit val configFormat   = yamlFormat2(Config)
   implicit val datasetReport  = yamlFormat2(DatasetReport)
   implicit val resultFormat   = yamlFormat5(RetirementReport)
+
+  implicit object TableYamlFormat extends YamlFormat[Table] {
+    def write(t: Table) = {
+      t match {
+        case d: DatedTable =>
+          YamlObject(
+            YamlString("name")              -> YamlString(d.name),
+            YamlString("storage_type")      -> YamlString(d.storage_type),
+            YamlString("expiration_column") -> YamlString(d.expiration_column),
+            YamlString("expiration_days")   -> YamlNumber(d.expiration_days),
+            YamlString("hold")              -> d.hold.toYaml,
+            YamlString("date_format_string") -> (if (d.date_format_string.isDefined) {
+                                                   YamlString(d.date_format_string.get)
+                                                 } else {
+                                                   YamlNull
+                                                 }),
+            YamlString("child_tables") -> d.child_tables.toYaml
+          )
+        case c: CustomTable =>
+          YamlObject(
+            YamlString("name")         -> YamlString(c.name),
+            YamlString("storage_type") -> YamlString(c.storage_type),
+            YamlString("filters")      -> c.filters.toYaml,
+            YamlString("hold")         -> c.hold.toYaml,
+            YamlString("child_tables") -> c.child_tables.toYaml
+          )
+      }
+    }
+
+    def read(value: YamlValue) = {
+      try {
+        value.asYamlObject.convertTo[DatedTable]
+      } catch {
+        case _: Throwable => value.asYamlObject.convertTo[CustomTable]
```

Comment:
Probably need to fix this, just realized if they actually run into another error I might want to specify that. I just couldn't think of a good way to determine if an inputted YamlValue is a dated table or not and thought of this.

Reply:
Yeah it should catch the 'ParsingException' or whatever and let others propagate.

Reply:
Put a TODO in there for now